apps/astarte_appengine_api/lib/astarte_appengine_api/queries.ex

```diff
@@ -88,46 +87,50 @@ defmodule Astarte.AppEngine.API.Queries do
     end
   end
 
-  def check_astarte_health(client, consistency) do
-    schema_statement = """
-      SELECT count(value)
-      FROM #{CQLUtils.realm_name_to_keyspace_name("astarte", Config.astarte_instance_id!())}.kv_store
-      WHERE group='astarte' AND key='schema_version'
-    """
+  def check_astarte_health(astarte_keyspace, consistency) do
+    schema_query =
+      from kv in KvStore,
+        prefix: ^astarte_keyspace,
+        where: kv.group == "astarte" and kv.key == "schema_version",
+        select: count(kv.value)
 
-    # no-op, just to check if nodes respond
-    # no realm name can contain '_', '^'
-    realms_statement = """
-      SELECT *
-      FROM #{CQLUtils.realm_name_to_keyspace_name("astarte", Config.astarte_instance_id!())}.realms
-      WHERE realm_name='_invalid^name_'
-    """
+    realm_query =
+      from Realm,
+        prefix: ^astarte_keyspace,
+        where: [realm_name: "_invalid^name_"]
 
-    schema_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(schema_statement)
-      |> DatabaseQuery.consistency(consistency)
-
-    realms_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(realms_statement)
-      |> DatabaseQuery.consistency(consistency)
-
-    with {:ok, result} <- DatabaseQuery.call(client, schema_query),
-         ["system.count(value)": _count] <- DatabaseResult.head(result),
-         {:ok, _result} <- DatabaseQuery.call(client, realms_query) do
+    opts = [consistency: consistency]
+
+    with {:ok, _result} <- safe_query(schema_query, opts),
+         {:ok, _result} <- safe_query(realm_query, opts) do
       :ok
     else
-      %{acc: _, msg: err_msg} ->
-        _ = Logger.warning("Health is not good: #{err_msg}.", tag: "db_health_check_bad")
+      err -> err
+    end
+  end
 
-        {:error, :health_check_bad}
+  defp safe_query(ecto_query, opts) do
+    {sql, params} = Repo.to_sql(:all, ecto_query)
 
-      {:error, err} ->
-        _ =
-          Logger.warning("Health is not good, reason: #{inspect(err)}.",
-            tag: "db_health_check_bad"
-          )
+    # Equivalent to a `Repo.all`, but does not raise if we get a Xandra Error.
+    case Repo.query(sql, params, opts) do
+      {:ok, result} ->
+        {:ok, result}
+
+      {:error, %Xandra.ConnectionError{}} ->
+        {:error, :database_connection_error}
+
+      {:error, :health_check_bad} ->
+        {:error, :database_connection_error}
+
+      {:error, %Xandra.Error{message: message, reason: reason}} ->
+        error_message =
+          case message do
+            "" -> inspect(reason)
+            message -> message
+          end
+
+        Logger.warning("Health is not good: #{error_message}", tag: "db_health_check_bad")
```

Comment:
`%Xandra.Error` implements the `Exception` behaviour. So
```suggestion
      {:error, %Xandra.Error{} = err}->
        Logger.warning("Health is not good: #{Exception.message(err)}", tag: "db_health_check_bad")
```